sops.go

```diff
@@ -189,6 +190,53 @@ func (branch TreeBranch) Set(path []interface{}, value interface{}) TreeBranch {
 	return set(branch, path, value).(TreeBranch)
 }
 
+func unset(branch interface{}, path []interface{}) (interface{}, error) {
+	switch branch := branch.(type) {
+	case TreeBranch:
+		for i, item := range branch {
+			if item.Key == path[0] {
+				if len(path) == 1 {
+					branch = slices.Delete(branch, i, i+1)
+				} else {
+					v, err := unset(item.Value, path[1:])
+					if err != nil {
+						return nil, err
+					}
+					branch[i].Value = v
+				}
+				return branch, nil
+			}
+		}
+		return nil, fmt.Errorf("Key not found: %s", path[0])
+	case []interface{}:
+		position := path[0].(int)
```

Comment:
This panics if `path[0]` happens to be a string. It's the same as in `set()` though, so I would probably keep it as-is for now, but this is something we should fix at some point (i.e. return a useful error message instead of panicking).

Reply:
Agreed. I'll leave it for another PR then.